### ISSUE TITLE
World leak server side.

### ISSUE THREAD

**Issue by MrMadakey** [XCompWiz/Mystcraft-Issues#88]
So I tweeted at you about it and previously I started up a server and it was using obscene amounts of memory on the server side. So I started doing a little bit of research after finding multiple world leaks inside my server. I downgraded the mod pack down 1.3.4 Infinity and there was a couple of crucial mods missing. However it was the only mod pack stable to run so I did a little more research and asked a couple of people and it turned out the only fix was to disable Mystcraft. Now you would think it was because I had multiple worlds created however you're missing one crucial thing.... We had literally just started on the world and hadn't even started on Mystcraft... Now I have installed something known as 1.5.1 ch fix server side. The client was FTB infinity 1.7 (v1.5.1) Minecraft version 1.7.10

**Comment by XCompWiz** [comment]
I'm assuming you verified that removing Mystcraft DID fix the issue?
I'd need some actual profiling data to be able to determine what the problem is.  I can trawl my code, but I don't think I keep any world instances anywhere...  And if you hadn't started Mystcraft then I don't know why Mystcraft would be tracking any world references.

Can you tell which worlds were leaking?  If you hadn't started Mystcraft, how were you traveling to the other worlds?

**Comment by MrMadakey** [comment]
I hadn't traveled to the other worlds at all though that was the problem. The only reason I found out there was a memory leak was because the server was running over time for no reason on the latest snap shot and I went to go consult help about it from some helpful people at creeperhost. They told me to install a 1.5.1 fix which completely disables Mystcraft so the first thing I did was come here. If I was to replicate the issue it could well cause issues for the server or my world.

**Comment by XCompWiz** [comment]
Sounds like a prescribed solution to fix a problem which may or may not have anything to do with Mystcraft.  If it cannot be replicated, I can't begin to track it down.
Further, if you haven't been to any other dimensions, you can have worlds leak...  Mystcraft won't load any worlds until you travel to them, aside from the profiling age which it doesn't save and discards when necessary. There were some issues with that, but they have been fixed some time ago.  I don't know what version of Infinity you are on.

I don't have anything to go off of here.  As far as I can tell, someone just blamed Mystcraft for something.  It gets that a lot, as it's a complex mod and exposes a number of underlying issues in places which don't turn up often elsewhere.  If you hadn't used any Mystcraft content or changed dimensions, Mystcraft shouldn't be doing anything (except profiling...).
I don't know what version of Mystcraft Infinity 1.5.1 is running, and since the site is rather crap you'll have to tell me. If it's recent enough it could be profiling. If it's old enough it might be keeping those chunks loaded, but they should all go away when done.  If it's just the profiling you are having issues with then once it's done it will all go away, it just takes 15 mins at most under normal circumstances.
If it's profiling, though, it should NOT be leaking worlds. Should that be the case then something more problematic is going on, as someone is messing with the profiling dimension being loaded.

tl;dr:
In order to do anything at all I'll need at least a log from whatever the original issue was. The full server log.

**Comment by RoqueDeicide** [comment]
Well, [this site](https://feedthebeast.atlassian.net/wiki/display/PML/Infinity+1.7+1.5.1) says that FTB Infinity 1.5.1 uses Mystcraft 0.11.5.00

I'd like to add a little: a while ago I tried using Mystcraft 0.11.7 on my modpack and while profiling was going, everything was fine, memory usage was normal, fps and tps were also normal (not sure about tps), however as soon as the profiling was over that's when the heap usage would blow up. And after that whenever I restarted the server, the profiling wouldn't start but the heap was already blown up. It was a few weeks ago and I downgraded to Mystcraft 0.11.0 to fix the issue for myself, so my memory might be a little rusty.

My setup had 2 void ages (1 of them not used at all), and I have RFTools mod with 1 dimension created with it (also not used).

**Comment by XCompWiz** [comment]
Interesting, so the problems only started after profiling stopped?  Again, if you could get a log of the server when that happened I might be able to help.

**Comment by XCompWiz** [comment]
OK, I think that, with the help of slowpoke, I've tracked this down. Hopefully the next release will significantly improve performance in this area.  I'll be releasing a patched build sometime today.

**Comment by XCompWiz** [comment]
Any news on if Monday's release fixed it?

**Comment by RoqueDeicide** [comment]
Played one SSP session with the new Mystcraft, everything was OK, profiling didn't affect much and after it nothing went wrong, heap usage was completely normal.
